lib/puppet/util/windows/registry.rb

```diff
@@ -220,6 +220,16 @@ def reg_query_info_key_max_lengths(key)
     def read(key, name_ptr, *rtype)
       result = nil
 
+      string_sanitizer = lambda do |value|
```

Comment:
Recreating an anonymous function every time `read` is call is not necessary here.

Reply:
I think what you have is more complicated than necessary. I think a private method in this class like this should be sufficient:

```
private
def sanitize(str)
  # Replace byte sequences that do not compose
  # to valid UTF8, and null bytes with the Unicode
  # Replacement Character: "�"
  value.scrub!.gsub!(/\0/, "\uFFFD")
end
```

This is because:

* We don't want to recreate an anonymous function for each read
* `value` should always be a string, so you shouldn't need to check if it has a `gsub` method
* we can only run on Ruby 2.4+ where strings have a `scrub` method